python/py-kcs11/Portfile

```diff
@@ -15,17 +15,20 @@ description         PKCS11 wrapper for Python
 long_description    PyKCS11 let you access to almost all PKCS#11 functions and data types \
                     using any PKCS#11 library, such as the various modules supplied by \
                     smartcard vendors.
-
-checksums           rmd160  05cf49e68caaa2ab43049d3f6757347427edd7c1 \
-                    sha256  b09cf6609231061f52bc888237c06353875d84ee9ff407a4ad6e987aad99b8b5 \
-                    size    77133
+homepage            https://github.com/LudovicRousseau/PyKCS11
+distname            pykcs11-${version}
+checksums           rmd160  aba82c1130e8e85a49a9a5a4ff0a0bcc58f32d6a \
+                    sha256  7b667e959fa0b6ad0750b03e2081a059cbe9a62789766143e50f90988a338994 \
+                    size    82675
 
 patchfiles          patch-setup.py-no-usr-local.diff
 
-python.versions     38
+python.versions     39 310 311 312
+python.pep517       no
```

Comment:
ideally we don't want to set this - what fails?

Reply:
I think it was not finding setuptools, and that was on Sonoma, so no powerpc oddities )

Reply:
that doesn't sound very plausible as the default backend would add it. Eventually things will break if they disable pep517... I can check later.

Reply:
Ok, this is on 10.8.5, what I have at hand. `swig` is not found:
```
--->  Configuring py312-kcs11
--->  Building py312-kcs11
Executing:  cd "/opt/local/var/macports/build/_opt_MountainLionPorts_python_py-kcs11/py312-kcs11/work/pykcs11-1.5.17" && /opt/local/Library/Frameworks/Python.framework/Versions/3.12/bin/python3.12 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_opt_MountainLionPorts_python_py-kcs11/py312-kcs11/work 
* Getting build dependencies for wheel...
running egg_info
writing PyKCS11.egg-info/PKG-INFO
writing dependency_links to PyKCS11.egg-info/dependency_links.txt
writing top-level names to PyKCS11.egg-info/top_level.txt
file PyKCS11/LowLevel.py (for module PyKCS11.LowLevel) not found
reading manifest file 'PyKCS11.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'COPYING'
writing manifest file 'PyKCS11.egg-info/SOURCES.txt'

ERROR Missing dependencies:
	swig
Command failed:  cd "/opt/local/var/macports/build/_opt_MountainLionPorts_python_py-kcs11/py312-kcs11/work/pykcs11-1.5.17" && /opt/local/Library/Frameworks/Python.framework/Versions/3.12/bin/python3.12 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_opt_MountainLionPorts_python_py-kcs11/py312-kcs11/work 
Exit code: 1
Error: Failed to build py312-kcs11: command execution failed
Error: See /opt/local/var/macports/logs/_opt_MountainLionPorts_python_py-kcs11/py312-kcs11/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port py312-kcs11 failed
```
Removing `pep517` fixes this. I have no idea why.

Reply:
It needs "swig" to build; the only reason it bails out early with pep517 because it notices the dependency is missing. The "setup.py" checks for the presence of swig and throws an error if not found; that seems to be more lenient for some reason. 

Regardless, the error message is real; you should fix the dependencies and remove the line.

Reply:
But it already has `swig-python` there, what else does it need?

Reply:
Ok done.